### PR TITLE
Replace kapt with ksp and update dependencies

### PR DESCRIPTION
**Pull request by simophin**
In order to use KSP, I have to:
1. Update kotlin version to the 1.9.x series (updating to 2.0 seems a bit risky at this moment)
2. Update compose compiler to match the kotlin update
3. Deprecated generated APIs aren't usable anymore in Glide so I followed the official guide to do the replacement (https://bumptech.github.io/glide/doc/generatedapi.html)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -14,19 +14,19 @@
 android.enableJetifier=true
 
 org.gradle.jvmargs=-Xmx2048M -Dkotlin.daemon.jvm.options\="-Xmx2048M"
-org.gradle.unsafe.configuration-cache=true
 
 gradlePluginVersion=7.3.1
 googleServicesVersion=4.3.12
-kotlinVersion=1.8.21
+kotlinVersion=1.9.24
```
Comment (AL-Session) [inline gradle.properties:20]: We don't want 1.9.25 or the newly stable 2.0.0?
https://kotlinlang.org/docs/releases.html#release-details

**Review by AL-Session — APPROVED** [review]
Looking good! We could probably push to Kotlin version 1.9.25 if we didn't want to make the jump to 2.0.0 yet - will leave that decision up to you & the gang. Approved!

File: gradle.properties (hunk 1 above)
Comment (simophin) [reply gradle.properties:20]: The Kotlin version is tied by the Compose Compiler: https://developer.android.com/jetpack/androidx/releases/compose-kotlin, the latest non-2.0 version compose compiler supports is 1.9.24

To update Kotlin to 2.0 some other things need to be done, I would very much prefer doing those things later

Comment (AL-Session) [reply gradle.properties:20]: Ah, right - no worries - 1.9.24 it is, then! =D

**Review by ThomasSession — APPROVED** [review]

**Pull request merged**